### DISCUSSION TITLE
Estimated bill?

### DISCUSSION THREAD

**Discussion by Rhysgh** [BottlecapDave/HomeAssistant-OctopusEnergy#200] [Ideas]
Has anyone created an entity for the estimated bill?

**Comment by BottlecapDave** [comment]
Hello. I've not personally set one up (although might be a good plan), but you should be able to use a [utility meter](https://www.home-assistant.io/integrations/utility_meter/) on the previous cost entity. Please note that the utility meter states 

> A sensor tracking monthly usage will present accurate data starting the first of the next month after being added to Home Assistant.

You'll probably also want to set an offset of 1 day to account for the data being staggered.

**Reply by Rhysgh** [reply]
I have been playing with them to make something work, in doing so I see that we have all the data there that are needed to do this really well, unfortunately, I lack the Python skills to accomplish it. more than happy to help with refinement etc if you are interested in having a go?

**Reply by BottlecapDave** [reply]
Another option would be to setup an [input number](https://www.home-assistant.io/integrations/input_number/) entity, which you then add to and reset via two automations; The first automation would run every night and take the value from your `current_consumption` or `previous_consumption` sensors and add it to the input number sensor. The second automation would then run every day with a condition on it being on day 1 or 2 of the date, which would then be responsible for resetting the input number sensor.

**Reply by Rhysgh** [reply]
I have 2 helpers running one for import and one for export. 

Then I use this 

```
    # Estimated Octopus Bill #
    - name: "Estimated Octopus Bill"
      unique_id: f1c5e5a5-68f9-4a2c-bbaa-42cf2c4cc4b2
      icon: >
        mdi:currency-gbp
      state: >-
        {{ (states('sensor.import_calculator') | float - states('sensor.export_calculator') | float) | round(2) }}

      # Estimated Octopus ALL TIME FLUX #
    - name: "Estimated Octopus Bill All Time Flux"
      unique_id: ed2a051f-76ca-4b5c-a3d2-808dc8c2bd8f
      icon: >
        mdi:currency-gbp
      state: >-
        {{ states('sensor.import_calculator_all_time_flux') | float - states('sensor.export_calculator_all_time_flux') | float }}
```

and get this 

![image](https://user-images.githubusercontent.com/57257743/232841600-8fd7266f-373c-4ff1-a4fd-2f6e599d3a67.png)

it's Quick and dirty for me but actually, it's been really reliable
